nservicebus/sagas/concurrency.md

```diff
@@ -1,48 +1,50 @@
+## Changes to saga state
 
-In this case only one message is allowed to complete processing. The others roll back and the built-in retries in NServiceBus kick in. On the next retry, the saga instance is found, the race condition is solved, and that saga instance is updated instead, see below.
+When messages are handled concurrently by the same logical saga, and involve changing the saga state, NServiceBus ensures that only one message handling thread changes the state in the same transaction as receiving the message and sending any new messages.
 
-partial: unique
+When using [SQL persistence](/persistence/sql/saga-concurrency.md), the messages are handled in serial, one after another. In this case, all the message handling threads are allowed to succeed.
 
-## Concurrent access to existing saga instances
+For all other persisters, only one message handling thread is allowed to succeed. The saga state is updated, and sent messages are dispatched. The other message handling threads fail, roll back, and enter [recoverability](/nservicebus/recoverability/). When those messages are retried, the process repeats. One or more of those messages may attempt to complete the saga. See below for how that scenario is handled.
 
-When messages concurrently tries to update the same saga instance the storage will either detect and throw a concurrency exception or serialize access to the instance. Concurrency exceptions will be automatically resolved by the NServiceBus retries will.
+## Completing a saga
 
-Another option is to use a [transaction isolation level](https://msdn.microsoft.com/en-us/library/system.transactions.isolationlevel.aspx) of serializable but that causes [excessive locking](https://docs.microsoft.com/en-us/sql/t-sql/statements/set-transaction-isolation-level-transact-sql) with considerable performance degradation.
+When messages are handled concurrently by the same logical saga, and the handling of at least one message involves completing the saga, NServiceBus ensures that only one message handling thread completes the saga. This is done in the same transaction as receiving the message and sending any new messages.
 
-NOTE: While `Serializable` is the default isolation level for TransactionScopes, in NServiceBus Version 4 and higher the isolation level will default to `ReadCommitted`.
+When using [SQL persistence](/persistence/sql/saga-concurrency.md), the messages are handled in serial, one after another.
 
+For all other persisters, only one message handling thread is allowed to succeed. The saga state is deleted, and sent messages are dispatched. The other message handling threads fail, roll back, and enter [recoverability](/nservicebus/recoverability/).
+
+In both cases, when subsequent messages are handled, the saga state is not found and the [messages are discarded](/nservicebus/sagas/saga-not-found.md).
 
 ### High load scenarios
```

Comment:
This isn't specific to Completing a saga, so I think it should be `##` instead of `###`.